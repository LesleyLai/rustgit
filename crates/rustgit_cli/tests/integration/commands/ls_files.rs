use crate::common::{git, populate_folder, rustgit};
use lazy_static::lazy_static;
use std::{fs, path::PathBuf};
use test_utils::TEST_DIR;

lazy_static! {
    static ref WORKING_DIR: PathBuf = {
        let working_dir = TEST_DIR.join("ls-files");
        fs::create_dir(&working_dir).unwrap();

        git(&working_dir).init();

        assert!(rustgit(&working_dir).ls_files().is_empty());

        populate_folder(&working_dir);

        assert!(rustgit(&working_dir).ls_files().is_empty());

        git(&working_dir).stage(&["."]);

        working_dir
    };
}

#[test]
fn files() {
    let working_dir = &WORKING_DIR;
    insta::assert_snapshot!(rustgit(&working_dir).ls_files());
}

#[test]
fn stage() {
    let working_dir = &WORKING_DIR;
    insta::assert_snapshot!(rustgit(&working_dir).ls_files_stage());
}
