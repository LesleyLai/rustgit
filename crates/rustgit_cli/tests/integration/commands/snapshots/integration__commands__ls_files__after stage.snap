---
source: crates/rustgit_cli/tests/integration/commands/ls_files.rs
assertion_line: 20
expression: rustgit().ls_files()
---
dir1/file_in_dir1_1
dir1/file_in_dir1_2
dir2/file_in_dir2_1
file1.txt
